main.tf

```diff
@@ -61,6 +79,29 @@ resource "azurerm_storage_account" "example" {
   location                 = azurerm_resource_group.example.location
   account_tier             = "Standard"
   account_replication_type = "GRS"
+
+  queue_properties {
+    hour_metrics {
+      version               = "1.0"
+      include_apis          = true
+      retention_policy_days = 365
+      enabled               = true
+    }
+    minute_metrics {
+      enabled               = true
+      version               = "1.0"
+      include_apis          = true
+      retention_policy_days = 365
+    }
+    logging {
+      delete                = true
```

Comment:
**Tenable.cs remediation**
 
**[MEDIUM]: Ensure that 'Auditing' is set to 'On' for Azure MSSQL Server**
 
**Vulnerability**
Auditing for MSSQL Server can help with monitoring for errors or anomalies, create a record for compliance requirements, as well as aid with performance tuning of the environment. Many compliance frameworks and regulatory requirements have a minimum retention period of 90 days as a standard. For more information on auditing in MSSQL Server, see the Azure documentation.
References:
[https://learn.microsoft.com/en-us/azure/azure-sql/database/auditing-overview?view=azuresql](https://learn.microsoft.com/en-us/azure/azure-sql/database/auditing-overview?view=azuresql)
 
**Remediation**
In Azure Console - 
1. In Azure portal, navigate to Auditing under Security heading in MSSQL database or MSSQL server pane.
2. Click on enable auditing switch to enable it, more configuration options are available through 'view server settings' link.

In Terraform -
Current Azure Provider versions:
1. For each azurerm_mssql_server resource, create an azurerm_mssql_database_extended_auditing_policy resource.

Azure Provider versions prior to 2.86.0:
1. In the azurerm_mssql_server resource, create an extended_auditing_policy block to configure retention days and other parameters for auditing policy.

References:
[https://learn.microsoft.com/en-us/azure/azure-sql/database/auditing-overview?view=azuresql](https://learn.microsoft.com/en-us/azure/azure-sql/database/auditing-overview?view=azuresql)
[https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_database_extended_auditing_policy](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_database_extended_auditing_policy)
[https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_server](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_server)
[https://registry.terraform.io/providers/hashicorp/azurerm/2.85.0/docs/resources/mssql_server](https://registry.terraform.io/providers/hashicorp/azurerm/2.85.0/docs/resources/mssql_server)

source: tenablecs-